Build error for nokogiri 1.14.5 with GCC 13 on RISC-V

<!--
  Please fill out each section below, otherwise, your issue will be closed. This info allows Metasploit maintainers to diagnose (and fix!) your issue as quickly as possible.

  Useful Links:
  - Wiki: https://docs.metasploit.com/
  - Reporting a Bug: https://docs.metasploit.com/docs/using-metasploit/getting-started/reporting-a-bug.html

  Before opening a new issue, please search existing issues: https://github.com/rapid7/metasploit-framework/issues
-->

## Steps to reproduce

1. Build following the [Setting Up a Metasploit Development Environment](https://docs.metasploit.com/docs/development/get-started/setting-up-a-metasploit-development-environment.html) page, with GCC 13 on Arch Linux RISC-V

## Were you following a specific guide/tutorial or reading documentation?

[Setting Up a Metasploit Development Environment](https://docs.metasploit.com/docs/development/get-started/setting-up-a-metasploit-development-environment.html)

## Expected behavior

Build successfully.

## Current behavior

```text
test_global_handlers.c: In function ‘rb_foreign_error_handler’:
test_global_handlers.c:24:35: error: passing argument 2 of ‘xmlSetStructuredErrorFunc’ from incompatible pointer type
[-Wincompatible-pointer-types]
   24 |   xmlSetStructuredErrorFunc(NULL, foreign_error_handler);
      |                                   ^~~~~~~~~~~~~~~~~~~~~
      |                                   |
      |                                   void (*)(void *, xmlError *) {aka void (*)(void *, struct _xmlError *)}
In file included from /usr/include/libxml2/libxml/valid.h:15,
                 from /usr/include/libxml2/libxml/parser.h:19,
                 from ./nokogiri.h:35,
                 from test_global_handlers.c:1:
/usr/include/libxml2/libxml/xmlerror.h:898:57: note: expected ‘xmlStructuredErrorFunc’ {aka ‘void (*)(void *, const struct _xmlError *)’} but
argument is of type ‘void (*)(void *, xmlError *)’ {aka ‘void (*)(void *, struct _xmlError *)’}
  898 |                                  xmlStructuredErrorFunc handler);
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
make: *** [Makefile:246: test_global_handlers.o] Error 1
```

I saw [nokogiri is locked](https://github.com/rapid7/metasploit-framework/blob/422acd3145a295a8f3a839495385ffda84668048/metasploit-framework.gemspec#L86-L87) to '~> 1.14.0'. Wondering is there any way to upgrade since nokogiri fixed the error in https://github.com/sparklemotion/nokogiri/commit/cdc2150ce847dec14f54d7602667867f1e6702ab, released since 1.16.0.

### Metasploit version

Get this with the `version` command in msfconsole (or `git log -1 --pretty=oneline` for a source install).

```text
Framework: 6.4.7-dev
Console  : 6.4.7-dev
```

--- a/metasploit-framework.gemspec
+++ b/metasploit-framework.gemspec
@@ -82,9 +82,8 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency 'network_interface'
   # NTLM authentication
   spec.add_runtime_dependency 'rubyntlm'
-  # Needed by anemone crawler
-  # Locked until build env can handle newer version due to native compile issue in 1.15.x
-  spec.add_runtime_dependency 'nokogiri', '~> 1.14.0'
+  # Needed by for XML parsing
+  spec.add_runtime_dependency 'nokogiri'
   # Needed by db.rb and Msf::Exploit::Capture
   spec.add_runtime_dependency 'packetfu'
   # For sniffer and raw socket modules

